crawler/crawler.py

```diff
@@ -287,28 +288,50 @@ def crawler_worker(process_id, logfile, params):
 
     crawlutils.snapshot(**params)
 
+def docker_monitor(process_id, logfile, syncQ):
+    setup_logger('docker-monitor', logfile, process_id)
+    logger.info('*' * 50)
+    logger.info('Docker monitor #%d started.' % (process_id))
+    logger.info('*' * 50)
+
+    monitor = DockerMonitor(syncQ)
+    monitor.startMonitor()
 
 # (a) an autonomous push mode via command-line invocation
 
 def start_autonomous_crawler(num_processes, logfile):
 
     if params['crawlmode'] == 'OUTCONTAINER':
         jobs = []
+        syncQ = multiprocessing.JoinableQueue()
+        delList = multiprocessing.Manager().list()
 
         for index in xrange(num_processes):
             # XXX use options.get() instead
             options['partition_strategy']['name'] = 'equally_by_pid'
             partition_args = options['partition_strategy']['args']
             partition_args['process_id'] = index
             partition_args['num_processes'] = num_processes
+            partition_args['eventQ'] = syncQ
```

Comment:
would prefer that these arguments are not passed here. The reason is that ideally i want to eventually move all the options to snapshot() to proper python config files, and won't be able to do that if they are not all pure text.

Can you please move them to be extra (optional) options for snapshot().